### PR TITLE
api: get_trytes implementation

### LINKED ISSUE #428: getTrytes implementation

### PR DESCRIPTION
**Pull request by thibault-martinez**
Fixes #428 

# Test Plan:
Added test cases

### CODE REVIEW

File: ciri/api/api.c [hunk 1]
```diff
@@ -144,9 +95,39 @@ retcode_t iota_api_find_transactions(find_transactions_req_t const *const req,
...
+                           NUM_TRITS_SERIALIZED_TRANSACTION);
+    } else {
+      memset(tx_trytes, '9', NUM_TRYTES_SERIALIZED_TRANSACTION);
+    }
+    // TODO Remove when cclient uses hash containers
+    tx_trytes[NUM_TRYTES_SERIALIZED_TRANSACTION] = '\0';
+    get_trytes_res_add_trytes(res, tx_trytes);
+    nbr++;
+  }
+  if (nbr > api->max_get_trytes) {
```
Comment (th0br0) [inline ciri/api/api.c]: this check should happen inside or before the loop?

File: ciri/api/api.h [hunk 2]
```diff
@@ -23,8 +24,10 @@ typedef struct iota_api_s {
   thread_handle_t thread;
   bool running;
   uint16_t port;
+  tangle_t *tangle;
   serializer_t serializer;
   serializer_type_t serializer_type;
+  size_t max_get_trytes;
```
Comment (th0br0) [inline ciri/api/api.h]: could we encode this in some `api_limits_t` struct? there's also other limits on the API

File: ciri/api/tests/test_api.c [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+#include "consensus/test_utils/bundle.h"
+#include "consensus/test_utils/tangle.h"
+
+static char *test_db_path = "ciri/api/tests/test.db";
+static char *ciri_db_path = "ciri/api/tests/ciri.db";
+static connection_config_t config;
+static iota_api_t api;
+static tangle_t tangle;
+
+void setUp(void) {
```
Comment (th0br0) [inline ciri/api/tests/test_api.c]: you're missing a test for the max_get_trytes limit?

**Review by th0br0 — CHANGES_REQUESTED** [review]

File: ciri/api/api.c (hunk 1 above)
Comment (thibault-martinez) [reply ciri/api/api.c]: Absolutely

File: ciri/api/tests/test_api.c (hunk 3 above)
Comment (thibault-martinez) [reply ciri/api/tests/test_api.c]: No it's there

File: cclient/request/get_trytes.c [hunk 4]
```diff
@@ -7,11 +7,24 @@
...
+  get_trytes_req_t* req = (get_trytes_req_t*)malloc(sizeof(get_trytes_req_t));
+  if (req) {
+    req->hashes = flex_hash_array_new();
+  }
+  return req;
+}
+
+void get_trytes_req_free(get_trytes_req_t** const req) {
+  if (req) {
+    flex_hash_array_free((*req)->hashes);
```
Comment (th0br0) [inline cclient/request/get_trytes.c]: could we restructure this similar to
```
if(!req || !(*req)) return;

get_trytes_req_t* mat = *req;

if(mat->hashes) flex_hash_array_free(mat);
free(mat);
*req = NULL;
```

File: ciri/api/api.c [hunk 5]
```diff
@@ -144,9 +95,37 @@ retcode_t iota_api_find_transactions(find_transactions_req_t const *const req,
...
+  flex_trit_t tx_trits[FLEX_TRIT_SIZE_8019];
+  tryte_t tx_trytes[NUM_TRYTES_SERIALIZED_TRANSACTION + 1];
+  DECLARE_PACK_SINGLE_TX(tx, txp, pack);
+
+  if (flex_hash_array_count(req->hashes) > api->limits.max_get_trytes) {
+    return RC_API_MAX_GET_TRYTES;
+  }
+  LL_FOREACH(req->hashes, iter) {
+    hash_pack_reset(&pack);
+    if ((ret = iota_tangle_transaction_load(api->tangle, TRANSACTION_FIELD_HASH,
```
Comment (th0br0) [inline ciri/api/api.c:113]: Could you add a comment here that implies that we need to consider concurrency here?

**Review by th0br0 — CHANGES_REQUESTED** [review]

File: ciri/api/api.c (hunk 5 above)
Comment (thibault-martinez) [reply ciri/api/api.c:113]: Due to statements ?

Comment (th0br0) [reply ciri/api/api.c:113]: Yeah...

**Review by th0br0 — APPROVED** [review]

**Pull request merged**